db/db.go

```diff
@@ -108,14 +113,22 @@ func getEnvOrDefault(varName string) string {
 // LoadEnvironmentVariables loads relevant environment variables into a
 // Config object.
 func LoadEnvironmentVariables() (Config, error) {
-	return Config{
+	config := Config{
 		Port:          getEnvOrDefault("PORT"),
-		DbHost:        getEnvOrDefault("DB_HOST"),
-		DbName:        getEnvOrDefault("DB_NAME"),
-		DbUsername:    getEnvOrDefault("DB_USERNAME"),
-		DbPass:        getEnvOrDefault("DB_PASSWORD"),
 		DbTokenTable:  getEnvOrDefault("DB_TOKEN_TABLE"),
 		DbDomainTable: getEnvOrDefault("DB_DOMAIN_TABLE"),
 		DbScanTable:   getEnvOrDefault("DB_SCAN_TABLE"),
-	}, nil
+	}
+	if flag.Lookup("test.v") != nil {
+		config.DbHost = getEnvOrDefault("TEST_DB_HOST")
+		config.DbName = getEnvOrDefault("TEST_DB_NAME")
+		config.DbUsername = getEnvOrDefault("TEST_DB_USERNAME")
+		config.DbPass = getEnvOrDefault("TEST_DB_PASSWORD")
+	} else {
+		config.DbHost = getEnvOrDefault("DB_HOST")
+		config.DbName = getEnvOrDefault("DB_NAME")
+		config.DbUsername = getEnvOrDefault("DB_USERNAME")
+		config.DbPass = getEnvOrDefault("DB_PASSWORD")
+	}
```

Comment:
Similar question to above-- could we have the flag force the scanner to use a different `.env` file?

Reply:
IMO the value of dotenv if that it can take config either from a file or from the environment. When we build the `app` docker container we don't include the `.env` file - instead we load its contents into the container's environment using the `env_file` key in the compose file. 

I'm not sure it will actually matter for deploys of this app, but we normally try to follow the twelve-factor app guidelines, which [require loading config from the environment](https://12factor.net/config).

If this code seems a bit messy, maybe we could pass a boolean to `getEnvOrDefault` indicating whether to prefix an env var with `TEST` if the test flag is set.

I do like having a separate default for TEST_DB_NAME so that users will avoid wiping out their database if they accidentally run tests without the test vars set.

Reply:
It is kinda appealing though...maybe we could allow in the normal env in the environment and the keep the test env in a file when the normal env needs to be overridden. I'll play with it a bit and see how it looks.

Reply:
Instead of providing a `.env.test` then, if people are expected to run tests via `docker-compose`, could we provide a different compose file?

> I do like having a separate default for TEST_DB_NAME so that users will avoid wiping out their database if they accidentally run tests without the test vars set.

Good point! Totally agree here.

Another question I had about the flag but evidently forgot to mention in the original review: is `test.v` set when `-v` is given to the `test` command?

Reply:
I went with your original suggestion of loading a `.env.test` file during tests and loading `.env` during normal runs. Either way, config that's specified directly on the environment will be included, I think we're still good for twelve-factor-ness. I used `dotenv.Overload` to override the envionment for test runs in the Docker container.

> Another question I had about the flag but evidently forgot to mention in the original review: is test.v set when -v is given to the test command?

I wasn't able to find documentation for this, but testing locally it runs regardless of whether the `-v` flag is given, while `flag.Lookup("test")` doesn't appear to work at all.